core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -344,21 +341,35 @@ private fun computeIndicationBrakingCurveFromRef(
 
 /**
  * Keep the part of the full braking curve which is located underneath the overlay and intersects
- * with it or with begin position. If the part has no intersection, return null. Ex: the part only
- * has negative positions, and the overlay starts at 0.0.
+ * with it or with begin position. The curve should always intersect, throw otherwise.
  */
 private fun keepBrakingCurveUnderOverlay(
-    fullBrakingCurve: EnvelopePart,
+    fullBrakingCurve: Envelope,
     overlay: Envelope,
     beginPos: Double
-): EnvelopePart? {
-    if (fullBrakingCurve.endPos <= beginPos) {
-        return null
-    }
-    val positions = fullBrakingCurve.clonePositions()
-    val speeds = fullBrakingCurve.cloneSpeeds()
-    val timeDeltas = fullBrakingCurve.cloneTimes()
-    val nbPoints = fullBrakingCurve.pointCount()
+): EnvelopePart {
+    if (fullBrakingCurve.endPos <= beginPos)
+        throw IllegalArgumentException(
+            "Expected fullBrakingCurve position range to intersect with [beginPos, +infinity[."
```

Comment:
Why not rejecting already if there is only one position?
```suggestion
    if (fullBrakingCurve.endPos < beginPos)
        throw IllegalArgumentException(
            "Expected fullBrakingCurve position range to intersect with ]beginPos, +infinity[."
```

Reply:
Technically, none, it woulda just thrown later. Anyhow, this code disappears since I will be removing these exceptions ;)

Reply:
As discussed, the code was good, not the comment (and I only read the comment 😅).
And agreed and reviewed to remove the final check 👍